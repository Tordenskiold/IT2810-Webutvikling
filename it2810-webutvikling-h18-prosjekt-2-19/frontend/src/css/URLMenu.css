.submitButton {
  position: fixed;
  bottom: 30px;
  right: 30px;
  height: 100px;
  width: 100px;
  background-color: #de5833;
  border-radius: 30%;
  border: black;
  align-items: center;
  color: white;
  font-size: 50px;
  cursor: pointer;
}

.paper {
  width:650px;
  background-color: white;
  border-radius: 10px;
  margin: auto;
  border-width: 10px;
  height: auto;
  outline: 0;
  box-shadow: black;
  -webkit-box-shadow: 0px 1px 20px 0px rgba(0, 0, 0, 0.41);
  -moz-box-shadow: 0px 1px 20px 0px rgba(0, 0, 0, 0.41);
  box-shadow: 0px 1px 20px 0px rgba(0, 0, 0, 0.41);
}

.modalButton {
  height: 40px;
  width: 120px;
  margin-left: 60px;
  background-color: #de5833;
  outline: 0;
  font-size: 20px;
  color: white;
  cursor: pointer;
  border-radius: 5%;
  margin-bottom: 30px;
}

.modalInput {
  background-color: rgba(168, 167, 167, 0.466);
  margin-left: 15px;
  width: 450px;
  height: 30px;
  border-radius: 3%;
  border-color: #c73f1a;
  border-width: 1px;
  align-content: flex-start;
  outline: 0;
  margin-top: 30px;
}

.label {
  font-size: 15px;
  font-weight: bold;
  align-items: center;
  padding-left: 13%;
}

.buttonDiv {
  margin-top: 50px;
  margin-left: 30%;
  width: 200px;
}

.form {
  margin-top: 100px;
  margin-left: 10%;
}

.format {
  margin-top: 10px;
  font-size: 12px;
  margin-left: 13%;
}

.test {
  margin-left: 35%;
  margin-top: 30%;
  padding-top: 1%;
}
