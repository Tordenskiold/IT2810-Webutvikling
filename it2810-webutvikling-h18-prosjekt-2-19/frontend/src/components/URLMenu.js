import React from 'react';
import Modal from '@material-ui/core/Modal';
import axios from 'axios';
import '../css/URLMenu.css';

class URLMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = { URL: ' ', open: false };
  }

  handleChange = (event) => {
    this.setState({ URL: event.target.value }, () => {});
  };

  //function for adding the url to the database
  handleClick = () => {
    const { URL } = this.state;
    const { store } = this.props;
    if (!this.validateURL(URL)) {
      alert('The url you provided is not valid');
      this.handleClose();
    } else {
      axios
        .post('http://localhost:3000/api/entries/', {
          url: `https://${URL}`
        })
        .then((response) => {
          console.log('response');
          console.log(response);
        })
        .catch((error) => {
          console.log(error.response);

          store.getAuthors();
          store.databaseSearch();
          this.handleClose();
        });
      // alert(`URL SUBMITTED: ${URL}`);
    }
  };

//Closes pop-up form
  handleClose = () => {
    this.setState({ open: false });
  };
//Opens pop-up form
  handleOpen = () => {
    this.setState({ open: true });
  };

// valIdates the URL entered in the form
  validateURL = (inputUrl) => {
    if (
      inputUrl.split('/')[0] !== 'raw.githubusercontent.com'
      || !inputUrl.split('/')[1]
      || !inputUrl.split('/')[2]
      || !inputUrl.split('/')[3]
      || inputUrl.split('/')[3] !== 'master'
      || !inputUrl.split('/')[4]
      || inputUrl.split('/')[4] !== 'README.md'
    ) {
      return false;
    }
    return true;
  };

//Renders the add button
//Creats the form for adding new urls to the database
  render() {
    const { open } = this.state;
    return (
      <div id="urldiv">
        <button type="button" onClick={this.handleOpen} className="submitButton">
          {open ? '-' : '+'}
        </button>
        <Modal id="modal" open={open} onClose={this.handleClose}>
          <div className="paper">
            <div className="test">
              <h1>Add new URL</h1>
            </div>
            <label className="label" id="2">
              URL:
              <input className="modalInput" type="text" name="name" onChange={this.handleChange} />
            </label>
            <div className="format">
              <label>Format: raw.githubusercontent.com/user/projectName/master/README.md</label>
            </div>
            <div className="buttonDiv">
              <input
                className="modalButton"
                type="submit"
                name="Submit"
                onClick={this.handleClick}
              />
            </div>
          </div>
        </Modal>
      </div>
    );
  }
}

export default URLMenu;
