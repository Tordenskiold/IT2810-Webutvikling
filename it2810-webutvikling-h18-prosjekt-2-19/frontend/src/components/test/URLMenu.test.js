import React from 'react';
import URLMenu from '../URLMenu';
import renderer from 'react-test-renderer';

//Jest test for URLMenu.js
//Creates a snapshot test which tests the rendering of the url menu

//snapshot test
describe("Testing URLMenu-component", () => {
  const component = renderer.create(<URLMenu />);
  const tree = renderer.create(<URLMenu />).getInstance();
  test("Renders correctly", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });



  describe("Testing functions", () => {
  
//Tests if the function closes the pop-up form
    test("test handleClose", () => {
      tree.handleClose();
      expect(tree.state.open).toBeFalsy();
    });

//Tests if the the URL is valid or not
    test("test validateURL", () => {
      const url = "test";
      expect(tree.validateURL(url)).toBeFalsy();
      const validurll = "raw.githubusercontent.com/user/prosject/master/README.md";
      expect(tree.validateURL(validurll)).toBeTruthy();
    });

  });
});
