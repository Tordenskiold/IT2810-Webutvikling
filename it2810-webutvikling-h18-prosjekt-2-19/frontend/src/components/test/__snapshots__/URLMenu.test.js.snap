// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing URLMenu-component Renders correctly 1`] = `
<div
  id="urldiv"
>
  <button
    className="submitButton"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
</div>
`;
